partner_firstname/models/base_config_settings.py

```diff
@@ -0,0 +1,57 @@
+# -*- coding: utf-8 -*-
+# © 2015 Antiun Ingenieria S.L. - Antonio Espinosa
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+from openerp import models, fields, api
+_logger = logging.getLogger(__name__)
+
+
+class BaseConfigSettings(models.TransientModel):
+    _inherit = 'base.config.settings'
+
+    partner_names_order = fields.Selection(
+        string="Partner names order",
+        selection="_partner_names_order_selection",
+        help="Order to compose partner fullname"
+    )
+
+    def _partner_names_order_selection(self):
+        return [
+            ('last_first', 'Lastname Firstname'),
+            ('last_first_comma', 'Lastname, Firstname'),
+            ('first_last', 'Firstname Lastname'),
+        ]
+
+    @api.multi
+    def get_default_partner_names_order(self):
+        return {
+            'partner_names_order': self.env['ir.config_parameter'].get_param(
+                'partner_names_order', 'last_first'),
+        }
+
+    @api.multi
+    def set_partner_names_order(self):
+        try:
+            self.env['ir.config_parameter'].set_param(
+                'partner_names_order', self.partner_names_order)
+        except:
+            self.env.cr.rollback()
+            self.env['ir.config_parameter'].set_param(
+                'partner_names_order', 'last_first')
+
+    @api.multi
+    def _partners_for_recalculating(self):
+        return self.env['res.partner'].search([
+            ('is_company', '=', False),
+            ('firstname', '!=', False), ('lastname', '!=', False),
+        ])
+
+    @api.multi
```

Comment:
Old api for actions methods is ```def action_my_button(self, cr, uid, ids, context=None):```, so AFAIK ```api.multi``` must be used in order to get all params encapsulated in self.